### DISCUSSION TITLE
RLS on PATCH not allowing data update

### DISCUSSION THREAD

**Discussion by Aging-Developer** [supabase/supabase#15811] [Questions] [answered]
I have a policy on an update to test table:  ((auth.jwt() ->> 'sub'::text) = (user_id)::text)

If RLS is enabled then the request (a PATCH made from postman) seems to pass the policy as there is no messages to state its not; however the data is not being updated, soon as i turn off RLS it works fine.  Is there anything else i need to do to get it to work and to be able to update the data?

p.s. (auth.uid() = user_id) doesn't seem to allow the data to be updated either, again no error messages.  I have double checked the JWT and the data in the user_id column.

Any help appreciated, i have been going mad trying to figure it out.

**Comment by GaryAustin1** [comment, marked as answer]
Do you meet your select policy?  Update will fail silently if you don't have access to the row.
Also you probably should show your actual policy and and what you are doing for the Authorization header and jwt you send in.
